File: app/src/main/java/com/softaai/designpatterns/java/creational/AbstractFactory/FactoryProducer.java
package com.softaai.designpatterns.java.creational.AbstractFactory;

import com.softaai.designpatterns.java.creational.AbstractFactory.color.ColorFactory;
import com.softaai.designpatterns.java.creational.AbstractFactory.shape.ShapeFactory;

/**
 * Created by Amol Pawar on 31-03-2019.
 * softAai Apps
 */
public class FactoryProducer {

    public static AbstractFactory getFactory(String choice){
        if(choice.equalsIgnoreCase("SHAPE")){
            return new ShapeFactory();
        }
        else if(choice.equalsIgnoreCase("COLOR")){
            return new ColorFactory();
        }
        return null;
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/creational/AbstractFactory/FactoryProducer.kt
package com.softaai.designpatterns.kotlin.creational.AbstractFactory

import com.softaai.designpatterns.kotlin.creational.AbstractFactory.color.ColorFactory
import com.softaai.designpatterns.kotlin.creational.AbstractFactory.shape.ShapeFactory


/**
 * Created by Amol Pawar on 01-04-2019.
 * softAai Apps
 */
object FactoryProducer {

    fun getFactory(choice:String): AbstractFactory? {

        if(choice.equals("Color", ignoreCase = true)){
            return ColorFactory()
        }
        else if(choice.equals("Shape", ignoreCase = true)){
            return ShapeFactory()
        }

        return null
    }
}